src/graph/GoExecutor.cpp

```diff
@@ -1104,44 +1169,78 @@ bool GoExecutor::processFinalResult(RpcResponse &rpcResp, Callback cb) const {
-                                                const std::string &prop) -> OptVariantType {
-                        auto dst = RowReader::getPropByName(&*iter, "_dst");
-                        if (!ok(dst)) {
-                            return Status::Error(
-                                folly::sformat("get prop({}.{}) failed", tag, prop));
-                        }
-                        auto vid = boost::get<int64_t>(value(std::move(dst)));
-
+                    getters.getDstTagProp = [&spaceId,
+                                             &dstId,
+                                             this] (const std::string &tag,
+                                                    const std::string &prop) -> OptVariantType {
                         TagID tagId;
                         auto found = expCtx_->getTagId(tag, tagId);
                         if (!found) {
                             return Status::Error(
                                     "Get tag id for `%s' failed in getters.", tag.c_str());
                         }
-
-                        if (saveTypeFlag) {
-                            SupportedType type = vertexHolder_->getType(vid, tagId, prop);
-                            colTypes.back() = type;
-                        }
-                        return vertexHolder_->get(vid, tagId, prop);
+                        return vertexHolder_->get(dstId, tagId, prop);
                     };
-                    getters.getVariableProp = [&saveTypeFlag, &colTypes, &vdata,
-                                               this](const std::string &prop) {
-                        if (saveTypeFlag) {
-                            colTypes.back() = getPropTypeFromInterim(prop);
-                        }
-                        return getPropFromInterim(vdata.get_vertex_id(), prop);
+                    getters.getVariableProp = [&srcId,
+                                               this] (const std::string &prop) {
+                        return getPropFromInterim(srcId, prop);
                     };
-                    getters.getInputProp = [&saveTypeFlag, &colTypes, &vdata,
-                                            this](const std::string &prop) {
-                        if (saveTypeFlag) {
-                            colTypes.back() = getPropTypeFromInterim(prop);
-                        }
-                        return getPropFromInterim(vdata.get_vertex_id(), prop);
+                    getters.getInputProp = [&srcId,
+                                            this] (const std::string &prop) {
+                        return getPropFromInterim(srcId, prop);
                     };
 
+                    std::unique_ptr<RowReader> reader;
+                    if (!edgeSchema.empty()) {
+                        auto it = edgeSchema.find(edgeType);
+                        DCHECK(it != edgeSchema.end());
+                        reader = RowReader::getRowReader(edge.props, it->second);
```

Comment:
because the reader depend edgeSchema, if the edgeSchema is empty, the reader is nullptr, then CHECK(reader != nullptr); will crash , so if can we return directly when edgeSchema is empty?

Reply:
If getAliasProp invoked, it means the reader should not be NULL.  Otherwise it is a bug.